### PR TITLE
Changed allocations for DM build

### PR DESCRIPTION
**Pull request by cnegre**
Changed allocations for DM build. Added some new routines in system.

---
This change is [Reviewable](https://reviewable.io/reviews/lanl/qmd-progress/114)

### CODE REVIEW

File: src/prg_system_mod.F90 [hunk 1]
```diff
@@ -254,17 +257,17 @@ subroutine prg_parse_system(system,filename,extin)
 
     if(.not.present(extin))then
        call prg_get_nameandext(filename,nametmp,extension)
-       filename = nametmp
     else
        extension = extin
+       nametmp = trim(filename)
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: You could also `trim(adjustl(filename))` to remove potentially leading whitespaces.

File: src/prg_system_mod.F90 [hunk 2]
```diff
@@ -254,17 +257,17 @@ subroutine prg_parse_system(system,filename,extin)
...
     endif
 
     select case(extension)
 
     case("xyz")
 
        !! For xyz format see http://openbabel.org/wiki/XYZ_%28format%29
-       io_name=trim(filename)//".xyz"
+       io_name=trim(nametmp)//".xyz"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: ditto

File: src/prg_system_mod.F90 [hunk 3]
```diff
@@ -312,7 +315,7 @@ subroutine prg_parse_system(system,filename,extin)
     case("pdb")
 
        !! For PDB format see http://www.wwpdb.org/documentation/file-format
-       io_name=trim(filename)//".pdb"
+       io_name=trim(nametmp)//".pdb"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: ditto

File: src/prg_system_mod.F90 [hunk 4]
```diff
@@ -359,14 +362,16 @@ subroutine prg_parse_system(system,filename,extin)
        allocate(system%coordinate(3,nats))
        allocate(system%mass(nats))
        allocate(system%lattice_vector(3,3))
+       allocate(system%resname(nats))
 
        system%lattice_vector = 0.0_dp
+       system%resname = "MOL"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: Why do you set `resname`? Isn't it overwritten when you read the file?

File: src/prg_system_mod.F90 [hunk 5]
```diff
@@ -422,7 +427,7 @@ subroutine prg_parse_system(system,filename,extin)
     case("ltt")
 
        !! For old inputblock.dat (old dat) format see the LATTE manual.
-       io_name=trim(filename)//".ltt"
+       io_name=trim(nametmp)//".ltt"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: `trim(adjustl(filename))`

File: src/prg_system_mod.F90 [hunk 6]
```diff
@@ -448,16 +453,14 @@ subroutine prg_parse_system(system,filename,extin)
...
-          ! ,system%coordinate(2,i),system%coordinate(3,i)
        enddo
 
        close(io_unit)
 
     case("dat")
 
        !! For new inputblock.dat (dat) format see the LATTE manual.
-       io_name=trim(filename)//".dat"
+       io_name=trim(nametmp)//".dat"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: ditto

File: src/prg_system_mod.F90 [hunk 7]
```diff
@@ -631,13 +634,15 @@ end subroutine prg_parse_system
   !! \param filename File name.
   !! \param extension Extension of the file.
   !!
-  subroutine prg_write_system(system,filename,extension)
+  subroutine prg_write_system(system,filename,extin)
     implicit none
-    character(*)                   ::  filename
+    character(len=*)               ::  filename
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: Nice

File: src/prg_system_mod.F90 [hunk 8]
```diff
@@ -695,17 +710,24 @@ subroutine prg_write_system(system,filename,extension)
           resindex = system%resindex
        endif
 
+       if(.not.allocated(system%resname))then
+          allocate(resname(nats))
+          resname = "MOL"
```
Comment (nicolasbock) [inline src/prg_system_mod.F90]: Same here, why set it to `MOL`?

File: src/prg_system_mod.F90 (hunk 1 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Changed

File: src/prg_system_mod.F90 (hunk 2 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Done

File: src/prg_system_mod.F90 (hunk 3 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Done

File: src/prg_system_mod.F90 (hunk 4 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Removed

File: src/prg_system_mod.F90 (hunk 5 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Done

File: src/prg_system_mod.F90 (hunk 6 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Done

File: src/prg_system_mod.F90 (hunk 7 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Changed for all of them

File: src/prg_system_mod.F90 (hunk 8 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: In this case I need it cause if by chance the system was read from .xyz format 
the the rename is not set. So, better to have something like "MOL"

Comment (nicolasbock) [reply src/prg_system_mod.F90]: Ah, thanks, I missed that.

File: src/prg_system_mod.F90 (hunk 5 above)
Comment (nicolasbock) [reply src/prg_system_mod.F90]: Thanks

File: src/prg_system_mod.F90 (hunk 7 above)
Comment (nicolasbock) [reply src/prg_system_mod.F90]: Thanks

**Pull request merged**